### PR TITLE
Format locale to Intl standard for translated date in notifications.

### LINKED ISSUE #11033: Bug of new version which caused by can't find the zh_CN locale
Reproduce steps: 
1. just login wallet
2. open wallet

Result:

image

image

Expect:
If it is not a bug, how should I avoid the error?

### PR DESCRIPTION
**Pull request by tmashuang**
Fixes: #11033

Explanation:  Format the incoming locale subtag from out list of translations to `Intl` standard for the date notification, found the full list [here](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).

Primarily this issue is with specific locales that are different subtag formats, for example `zh_CN` vs `zh-Hans-CN`/`zh-Hans` and `es_419` vs `es-419`/`419`.

Manual testing steps:  
The two conditions that need to be met are to have one of the non-Intl standard languages selected before reading/viewing the new What's new notification modal.

Don't know if the alternative of outright changing our own subtag locale folder would interfere with the initial browser language detection, or other ramifications.  If that method seems more appropriate, that can be achieved as well.

### CODE REVIEW

File: shared/notifications/index.js [hunk 1]
```diff
@@ -20,13 +20,31 @@ export const UI_NOTIFICATIONS = {
   },
 };
 
+const formatLocale = (locale) => {
```
[the remark this answers is not in the document]
Comment (danjm) [reply shared/notifications/index.js:23]: This is also done in `app/scripts/lib/get-first-preferred-lang-code.js`

[the remark this answers is not in the document]
Comment (danjm) [reply shared/notifications/index.js:23]: But why is the preferred language code being saved to state with the `_`?

[the remark this answers is not in the document]
Comment (danjm) [reply shared/notifications/index.js:23]: oh, we map from the version with the dash to the version with the `_`

[the remark this answers is not in the document]
Comment (danjm) [reply shared/notifications/index.js]: These are the results of different locale keys

Screenshot from 2021-05-11 13-04-23

[the remark this answers is not in the document]
Comment (tmashuang) [reply shared/notifications/index.js]: If the screenshot is accurate then the reproduction is flawed there there isn't an issue with those languages and all the users effected are incorrect.

**Review by Gudahtt — APPROVED** [review]
LGTM, good catch!

**Pull request merged**